src/ua_securechannel.h

```diff
@@ -29,36 +31,62 @@ struct ChunkEntry {
     UA_ByteString bytes;
 };
 
-/* For chunked responses */
-typedef struct {
-    UA_SecureChannel *channel;
-    UA_UInt32 requestId;
-    UA_UInt32 messageType;
-    UA_UInt16 chunksSoFar;
-    size_t messageSizeSoFar;
-    UA_Boolean final;
-    UA_StatusCode errorCode;
-} UA_ChunkInfo;
-
 struct UA_SecureChannel {
+    UA_Boolean temporary; // this flag is set to false if the channel is fully opened.
     UA_MessageSecurityMode  securityMode;
     UA_ChannelSecurityToken securityToken; // the channelId is contained in the securityToken
     UA_ChannelSecurityToken nextSecurityToken; // the channelId is contained in the securityToken
-    UA_AsymmetricAlgorithmSecurityHeader clientAsymAlgSettings;
-    UA_AsymmetricAlgorithmSecurityHeader serverAsymAlgSettings;
+    UA_AsymmetricAlgorithmSecurityHeader remoteAsymAlgSettings;
+    UA_AsymmetricAlgorithmSecurityHeader localAsymAlgSettings;
+
+    /** The active security policy and context of the channel */
+    const UA_SecurityPolicy *securityPolicy;
+    void *securityContext;
+
+    /** Stores all available security policies that may be used when establishing a connection. */
+    UA_SecurityPolicies availableSecurityPolicies;
```

Comment:
I just checked and the only way i see would be to pass the availablePolicies struct to UA_SecureChannel_processChunks (this gets called in UA_Server_processBinaryMessage, where we have access to the server struct where the policies are stored)
pass it on to UA_SecureChannel_processChunk and then to UA_SecureChannel_processAsymmetricOPNChunk
It seems ok i guess. Should i change it to this? That would eliminate the struct from permanently staying in a channel even though it is only needed on asymmetric OPN messages.
(If theses could be chunked we would need to pass it for each chunk, but i guess thats not a problem)